templates/contribute/git.md

```diff
@@ -0,0 +1,324 @@
+```
+
+This automatically sets up your fork as the `origin` remote, which is what we want.
+
+### Option B: If you already have mathlib4 cloned
+
+If you already have a clone from the original repository, you can reuse it. Just navigate to your existing mathlib4 directory:
+
+```bash
+cd path/to/your/existing/mathlib4
+```
+
+## Step 3: Set Up Remotes Correctly
+
+The remote setup depends on which option you chose above:
+
+### If you cloned your fork (Option A)
+
+You need to add the original repository as `upstream`:
+
+```bash
+git remote add upstream https://github.com/leanprover-community/mathlib4.git
+```
+
+### If you used an existing clone (Option B)
+
+You need to rename the existing remote and add your fork. Replace `YOUR_USERNAME` with your GitHub username:
+
+```bash
+git remote rename origin upstream
+git remote add origin https://github.com/YOUR_USERNAME/mathlib4.git
+```
+
+### Verify Your Remotes
+
+Regardless of which option you chose, verify your remotes are set correctly:
+
+```bash
+git remote -v
+```
+
+You should see:
+```
+origin    https://github.com/YOUR_USERNAME/mathlib4.git (fetch)
+origin    https://github.com/YOUR_USERNAME/mathlib4.git (push)
+upstream  https://github.com/leanprover-community/mathlib4.git (fetch)
+upstream  https://github.com/leanprover-community/mathlib4.git (push)
+```
+
+## Step 4: Configure the Master Branch
+
+First, fetch the branches from upstream:
+
+```bash
+git fetch upstream
+```
+
+Then make sure your `master` branch tracks `upstream/master`:
+
+```bash
+git branch --set-upstream-to=upstream/master master
+```
+
+## Step 5: Configure Git for Your Workflow
+
+### Set Default Push Behavior
+
+Configure git to push new branches to `origin` by default:
+
+```bash
+git config push.default current
+git config push.autoSetupRemote true
+```
+
+### Configure GitHub CLI (if installed)
+
+If you have the GitHub CLI installed, set the default repository to the upstream mathlib4:
+
+```bash
+gh repo set-default leanprover-community/mathlib4
+```
+
+This ensures that GitHub CLI commands like `gh pr checkout` will work with the main mathlib4 repository rather than your fork.
+
+### Prevent Accidental Commits to Master (Optional but Recommended)
+
+To avoid accidentally committing directly to `master`, you can set up a pre-commit hook. First, create the hook file:
+
+```bash
+mkdir -p .git/hooks
+cat > .git/hooks/pre-commit << 'EOF'
+#!/bin/sh
+branch="$(git rev-parse --abbrev-ref HEAD)"
+if [ "$branch" = "master" ]; then
+  echo "You can't commit directly to master branch"
+  exit 1
+fi
+EOF
+```
+
+Then make it executable:
+
+```bash
+chmod +x .git/hooks/pre-commit
+```
+
+---
+
+# Part 2: Daily Workflow
+
+These are the operations you'll use regularly when working on contributions.
+
+## Creating and Working on a New Branch
+
+### Keep Your Master Branch Up to Date
+
+**Do this before creating a new branch** to ensure you're working with the latest changes:
+
+```bash
+git switch master
+git pull
```

Comment:
I don't know much about git myself but I've been writing `git checkout master` for years. Is `switch` better?